docs/howtos/working_with_urls_and_api_endpoints.md

```diff
@@ -0,0 +1,423 @@
+# Working with URLs and API Endpoints
+
+This guide explains how to work with URLs and API endpoints in Kolibri, including the URL namespacing system used throughout the codebase.
+
+## Overview
+
+Kolibri uses a consistent URL namespacing pattern that bridges Django's backend URL system with JavaScript frontend code. This allows type-safe URL construction with automatic parameter validation across the entire stack.
+
+## URL Naming Convention
+
+All URLs in Kolibri follow the namespace pattern:
+
+```
+kolibri:<app_name>:<resource_name>_<endpoint>
+```
+
+### Core URLs
+
+Core Kolibri URLs use the `core` namespace:
+
+```
+kolibri:core:session_list
+kolibri:core:session_detail
+kolibri:core:facilityuser_list
+kolibri:core:facilityuser_detail
+kolibri:core:contentnode_tree_detail
+```
+
+### Plugin URLs
+
+Plugin URLs use the full plugin module path as the namespace:
+
+```
+kolibri:kolibri.plugins.coach:lessonreport_list
+kolibri:kolibri.plugins.learn:learnerlesson_detail
+kolibri:kolibri.plugins.device:devicepermissions_list
+```
+
+### REST Framework Endpoints
+
+Django REST Framework ViewSets automatically generate standardized URL names:
+
+- `<basename>_list` - List/create endpoint (GET/POST)
+- `<basename>_detail` - Detail/update/delete endpoint (GET/PUT/PATCH/DELETE)
+
+Custom actions on ViewSets create additional URL patterns based on the action name.
+
+## Backend: Defining URLs
+
+### Core App URLs
+
+Core API URLs are defined in `kolibri/core/auth/api_urls.py` (and similar files for other core apps):
+
+```python
+from rest_framework import routers
+
+router = routers.SimpleRouter()
+
+# This creates:
+# - kolibri:core:session_list
+# - kolibri:core:session_detail
+router.register(r"session", SessionViewSet, basename="session")
+
+# This creates:
+# - kolibri:core:facilityuser_list
+# - kolibri:core:facilityuser_detail
+router.register(r"facilityuser", FacilityUserViewSet, basename="facilityuser")
+```
+
+The `basename` parameter determines the URL name pattern. The namespace `kolibri:core` comes from:
+1. `app_name = "kolibri"` in `kolibri/core/urls.py`
+2. The tuple syntax in `core_urlpatterns = ([...], "core")`
+
+### Plugin URLs
+
+Plugin API URLs are defined in the plugin's `api_urls.py`:
```

Comment:
Again, this is a convention, not a requirement, and it also misses out the critical information of how to register these against the `KolibriPluginBase` derived plugin class.